src/MagnumPlugins/AssimpImporter/AssimpImporter.cpp

```diff
@@ -408,41 +456,51 @@ void AssimpImporter::doOpenData(Containers::Array<char>&& data, DataFlags) {
     }
 
     /* Fill hashmaps for index lookup for materials & textures */
+    size_t numTextures = 0;
+    for(aiMaterial* mat: Containers::arrayView(_f->scene->mMaterials, _f->scene->mNumMaterials))
+        for(aiMaterialProperty* property: Containers::arrayView(mat->mProperties, mat->mNumProperties))
+            if(Containers::StringView{property->mKey} == _AI_MATKEY_TEXTURE_BASE)
+                ++numTextures;
+    arrayReserve(_f->textures, numTextures);
+    arrayReserve(_f->images, numTextures);
+    _f->textureIndices.reserve(numTextures);
     _f->materialIndicesForName.reserve(_f->scene->mNumMaterials);
+
     aiString matName;
     UnsignedInt textureIndex = 0;
-    std::unordered_map<std::string, UnsignedInt> uniqueImages;
-    for(std::size_t i = 0; i < _f->scene->mNumMaterials; ++i) {
+    std::unordered_map<Containers::StringView, UnsignedInt> uniqueImages;
+    uniqueImages.reserve(numTextures);
+    for(UnsignedInt i = 0; i < _f->scene->mNumMaterials; ++i) {
         const aiMaterial* mat = _f->scene->mMaterials[i];
 
-        if(mat->Get(AI_MATKEY_NAME, matName) == AI_SUCCESS) {
-            std::string name = matName.C_Str();
-            _f->materialIndicesForName[name] = i;
-        }
+        if(mat->Get(AI_MATKEY_NAME, matName) == AI_SUCCESS)
+            _f->materialIndicesForName[matName] = i;
 
         /* Store first possible texture index for this material, next textures
            use successive indices. */
         _f->textureIndices[mat] = textureIndex;
-        for(std::size_t j = 0; j != mat->mNumProperties; ++j) {
+        for(UnsignedInt j = 0; j != mat->mNumProperties; ++j) {
```

Comment:
Reason I changed it is because I'd have to cast to `UnsignedInt` twice for the `arrayAppend(InPlaceInit)` onto `Pair`/`Triple` arrays to make it compile at all. I'll leave it as is, but let me know if you'd prefer the casts.

Reply:
Ah, right, then `UnsignedInt` is the way to go. Didn't see that myself, sorry. The code style advice in this case would then be "pick a type that's the least annoying to use" :)